Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -11378,9 +11382,94 @@ void CPhysicalGeometry::SetWallDistance(const CConfig *config, CADTElemClass *Wa
       unsigned long  elemID;
       int            rankID;
       su2double      dist;
+      su2double      localRoughness;
+      int index;
+
       WallADT->DetermineNearestElement(nodes->GetCoord(iPoint), dist, markerID, elemID, rankID);
+
       nodes->SetWall_Distance(iPoint, min(dist,nodes->GetWall_Distance(iPoint)));
+      index = GlobalMarkerStorageDispl[rankID]+ markerID;
+      localRoughness = GlobalRoughness_Height[index];
```

Comment:
You can use `auto` here instead of declaring the variables separately, again if possible skip this section if there are no rough markers.

Perhaps you could detect if roughness is needed in this function and call "SetGlobalMarkerRoughness" here before the wall distance loop, that way the method does not need to be virtual, it can be a private member function of CGeometry.